identity_data.go

```diff
@@ -0,0 +1,102 @@
+// Copyright 2016 Mender Software AS
+//
+//    Licensed under the Apache License, Version 2.0 (the "License");
+//    you may not use this file except in compliance with the License.
+//    You may obtain a copy of the License at
+//
+//        http://www.apache.org/licenses/LICENSE-2.0
+//
+//    Unless required by applicable law or agreed to in writing, software
+//    distributed under the License is distributed on an "AS IS" BASIS,
+//    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//    See the License for the specific language governing permissions and
+//    limitations under the License.
+package main
+
+import (
+	"bufio"
+	"bytes"
+	"strings"
+
+	// "github.com/mendersoftware/log"
+	"github.com/pkg/errors"
+)
+
+const (
+	identityDataHelper = "/usr/bin/mender-identity-data"
+)
+
+type IdentityDataGetter interface {
+	Get(cmdr Commander) (interface{}, error)
+}
+
+// Shorthand form of obtaining identity data with default runner
+func IdentityDataGet() (interface{}, error) {
+	r := IdentityDataRunner{}
+	return r.Get(&osCalls{})
+}
+
+type IdentityDataRunner struct {
+	Helper string
+}
+
+// Obtain identity data by calling a suitable helper tool
+func (id IdentityDataRunner) Get(cmdr Commander) (interface{}, error) {
+	helper := identityDataHelper
+
+	if id.Helper != "" {
+		helper = id.Helper
+	}
+
+	cmd := cmdr.Command(helper)
+	data, err := cmd.Output()
+
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to call %s", helper)
+	}
+
+	idata, err := parseIdentityData(data)
+	if err != nil {
+		return nil, errors.Wrapf(err, "failed to parse identity data")
+	}
+
+	return idata, nil
+}
+
+// device identity data content
+type IdentityData struct {
+	// device MAC address
+	MAC string `json:"mac"`
+}
+
+func parseIdentityData(data []byte) (interface{}, error) {
+	idata := IdentityData{}
```

Comment:
The example implementation as done in https://tracker.mender.io/browse/MEN-417 provides a key=value list of attributes.